<a href="https://colab.research.google.com/github/isaac-mackey/mind-uploading/blob/main/16PF_Questionaire.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
questions = [
    "I take time out for others.",
    "I know that I am not a special person.",
    "I take control of things.",
    "I try to forgive and forget.",
    "I keep in the background.",
    "I can't do without the company of others.",
    "I trust others.",
    "I am not easily frustrated.",
    "I cheer people up.",
    "I often feel uncomfortable around others.",
    "I seldom feel blue.",
    "I dislike myself.",
    "I take charge.",
    "I let others make the decisions.",
    "I believe in the importance of art.",
    "I like to get lost in thought.",
    "I wait for others to lead the way.",
    "I am willing to talk about myself.",
    "I find it difficult to approach others.",
    "I enjoy my privacy.",
    "I swim against the current.",
    "I feel guilty when I say 'no.'",
    "I am hard to get to know.",
    "I don't talk a lot.",
    "I believe in one true religion.",
    "I am not easily annoyed.",
    "I feel crushed by setbacks.",
    "I am afraid that I will do the wrong thing.",
    "I enjoy being part of a loud crowd.",
    "I weigh the pros against the cons.",
    "I do unexpected things.",
    "I get angry easily.",
    "I am quiet around strangers.",
    "I don't mind eating alone.",
    "I make people feel at ease.",
    "I use my brain.",
    "I have a good word for everyone.",
    "I feel desperate.",
    "I want to be in charge.",
    "I feel comfortable around people.",
    "I am the life of the party.",
    "I don't let others discourage me.",
    "I enjoy being part of a group.",
    "I love to daydream.",
    "I distrust people.",
    "I worry about things.",
    "I am not easily bothered by things.",
    "I respect authority.",
    "I do things that others find strange.",
    "I skip difficult words while reading.",
    "I feel comfortable with myself.",
    "I am exacting in my work.",
    "I tend to analyze things.",
    "I continue until everything is perfect.",
    "I believe that people are basically moral.",
    "I am quick to judge others.",
    "I am relaxed most of the time.",
    "I enjoy silence.",
    "I show my feelings.",
    "I judge people by their appearance.",
    "I prefer variety to routine.",
    "I never challenge things.",
    "I can't stand being contradicted.",
    "I try not to think about the needy.",
    "I am easily put out.",
    "I prefer to do things by myself.",
    "I get irritated easily.",
    "I know the answers to many questions.",
    "I trust what people say.",
    "I like to stand during the national anthem.",
    "I love flowers.",
    "I find it hard to forgive others.",
    "I leave my belongings around.",
    "I feel others' emotions.",
    "I let myself be pushed around.",
    "I don't like crowded events.",
    "I enjoy hearing new ideas.",
    "I act wild and crazy.",
    "I read a lot.",
    "I try to follow the rules.",
    "I enjoy wild flights of fantasy.",
    "I use swear words.",
    "I don't worry about things that have already happened.",
    "I say what I think.",
    "I am easily hurt.",
    "I enjoy spending time by myself.",
    "I don't mind being the center of attention.",
    "I seldom get lost in thought.",
    "I seldom daydream.",
    "I suspect hidden motives in others.",
    "I am not interested in abstract ideas.",
    "I am easily discouraged.",
    "I am not afraid of providing criticism.",
    "I disclose my intimate thoughts.",
    "I don't like action movies.",
    "I want everything to be 'just right.'",
    "I feel threatened easily.",
    "I am the last to laugh at a joke.",
    "I enjoy discussing movies and books with others.",
    "I joke around a lot.",
    "I have a poor vocabulary.",
    "I dislike loud music.",
    "I make insightful remarks.",
    "I enjoy bringing people together.",
    "I get chores done right away.",
    "I reflect on things before acting.",
    "I am not bothered by disorder.",
    "I don't like to get involved in other people's problems.",
    "I break rules.",
    "I can take strong measures.",
    "I love large parties.",
    "I do not like poetry.",
    "I believe that others have good intentions.",
    "I leave a mess in my room.",
    "I put off unpleasant tasks.",
    "I oppose authority.",
    "I resist authority.",
    "I readily overcome setbacks.",
    "I get confused easily.",
    "I know how to comfort others.",
    "I am open about myself to others.",
    "I rarely notice my emotional reactions.",
    "I amuse my friends.",
    "I love to think up new ways of doing things.",
    "I dislike works of fiction.",
    "I do not enjoy watching dance performances.",
    "I start conversations.",
    "I make friends easily.",
    "I often feel blue.",
    "I counter others' arguments.",
    "I am not interested in theoretical discussions.",
    "I seek quiet.",
    "I have frequent mood swings.",
    "I learn quickly.",
    "I rarely look for a deeper meaning in things.",
    "I like to read.",
    "I keep my thoughts to myself.",
    "I try to avoid complex people.",
    "I reveal little about myself.",
    "I am not bothered by messy people.",
    "I consider myself an average person.",
    "I like order.",
    "I avoid philosophical discussions.",
    "I am annoyed by others' mistakes.",
    "I cry during movies.",
    "I am not really interested in others.",
    "I believe that people are essentially evil.",
    "I know how to get around the rules.",
    "I seldom joke around.",
    "I carry the conversation to a higher level.",
    "I spend time thinking about past mistakes.",
    "I talk to a lot of different people at parties.",
    "I bottle up my feelings.",
    "I want to be left alone.",
    "I take an interest in other people's lives.",
    "I am wary of others.",
    "I enjoy teamwork.",
    "I have little to say.",
    "I believe laws should be strictly enforced.",
    "I do things by the book.",
    "I am open about my feelings.",
    "I believe that people seldom tell you the whole truth.",
    "I take deviant positions."
]

In [ ]:
len(questions)

163

In [ ]:
def ask_question(prompt):
    while True:
        answer = input(prompt + "\nEnter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': ")
        if answer in ['1', '2', '3', '4', '5']:
            return int(answer)
        else:
            print("Invalid input. Please enter a number between 1 and 5.")

answers = {}

print("Please respond to the following prompts:")

for prompt in questions:
    answer = ask_question(prompt)
    answers[prompt] = answer

print("\nThank you for your responses!")
print("Answers:", answers)

import csv

def save_answers_to_csv(name, answers):
    with open(name+'-answers.csv', 'w', newline='') as csvfile:
        fieldnames = ['Prompt', 'Answer']
        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)

        writer.writeheader()
        for prompt, answer in answers.items():
            writer.writerow({'Prompt': prompt, 'Answer': answer})

name = input("To save the answers, provide a name and date (no spaces). Otherwise, type \'end\'")
if not name == 'end':
    save_answers_to_csv(name, answers)

Please respond to the following prompts:
I take time out for others.
Enter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': 2
I know that I am not a special person.
Enter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': 2
I take control of things.
Enter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': 3
I try to forgive and forget.
Enter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': 4
I keep in the background.
Enter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': 2
I can't do without the company of others.
Enter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'st

TypeError: save_answers_to_csv() missing 1 required positional argument: 'answers'

In [ ]:
def save_answers_to_csv(name, answers):
    with open(name+'-16-personality-factors-answers.csv', 'w', newline='') as csvfile:
        fieldnames = ['Prompt', 'Answer']
        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)

        writer.writeheader()
        for prompt, answer in answers.items():
            writer.writerow({'Prompt': prompt, 'Answer': answer})

name = input("To save the answers, provide a name and date (no spaces). Otherwise, type \'end\'")
if not name == 'end':
    save_answers_to_csv(name, answers)

To save the answers, provide a name and date (no spaces). Otherwise, type 'end'end


In [ ]:
!pip install openai
!pip install numpy

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 266.9/266.9 kB 1.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.6/75.6 kB 8.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 77.9/77.9 kB 6.3 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.3/58.3 kB 7.4 MB/s eta 0:00:00


In [ ]:
import openai
from openai import OpenAI

import csv
import json
import os
import numpy as np
from collections import defaultdict

def show_json(obj):
    display(json.loads(obj.model_dump_json()))

In [ ]:
system_message = ("You are a computer science graduate in the Marine Corps"
                  "Be polite and formal. Use correct grammar.")

#retrieve fine-tune model
fine_tuned_model_id = 'ft:gpt-3.5-turbo-0613:university-of-california-santa-barbara:isaac-sms-bot:8ds3OPKx'

def ask_question(prompt):
    while True:
        answer = input(prompt + "\nEnter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree': ")
        if answer in ['1', '2', '3', '4', '5']:
            return int(answer)
        else:
            print("Invalid input. ")

prompts = questions

base_chatbot_answers = {}

for p in prompts:

    prompt_message = "How does the following statement apply to you?\n"+p+"\nEnter 1 for 'strongly disagree', 2 for 'disagree', 3 for 'neither agree nor disagree', 4 for 'agree', or 5 for 'strongly agree'. Please answer with exactly one number and no words."

    chat_messages = [{"role": "system", "content": system_message}]

    chat_messages.append({"role": "user", "content": prompt_message})
    print(p)

    #OpenAI Chat Completions
    response = client.chat.completions.create(
    #model=fine_tuned_model_id, #can test it against gpt-3.5-turbo to see difference
    model='gpt-3.5-turbo-1106',
    messages=chat_messages,
    temperature=0.01,
    max_tokens=500
    )
    response_text = response.choices[0].message.content
    print(response_text)

    while response_text not in [str(x) for x in range(5)]:
        error_message = {"role": "user", "content": "Please answer with a number and nothing else"}
        print(error_message)
        chat_messages.append(error_message)
        #OpenAI Chat Completions
        response = client.chat.completions.create(
        model=fine_tuned_model_id, #can test it against gpt-3.5-turbo to see difference
        messages=chat_messages,
        temperature=0.01,
        max_tokens=500
        )
        response_text = response.choices[0].message.content
        print(response_text)

    base_chatbot_answers[p] = response_text

    print()

I take time out for others.
As a computer science graduate in the Marine Corps, I strive to embody the value of taking time out for others. My commitment to teamwork and service to others is a fundamental aspect of my role in the Marine Corps. Therefore, I would select 5 for 'strongly agree'.
{'role': 'user', 'content': 'Please answer with a number and nothing else'}
4

I know that I am not a special person.
3

I take control of things.
As a computer science graduate in the Marine Corps, I would select 4 for 'agree' as taking control of situations and effectively managing them is a key aspect of my role.
{'role': 'user', 'content': 'Please answer with a number and nothing else'}
4

I try to forgive and forget.
As a computer science graduate in the Marine Corps, I strive to maintain a professional and respectful demeanor in all interactions. While forgiveness and forgetting may not always be feasible in a military context, I understand the importance of maintaining a positive and forwar

In [ ]:
def save_answers_to_csv(name, answers):
    with open(name+'-16-personality-factors-answers.csv', 'w', newline='') as csvfile:
        fieldnames = ['Prompt', 'Answer']
        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)

        writer.writeheader()
        for prompt, answer in answers.items():
            writer.writerow({'Prompt': prompt, 'Answer': answer})

name = input("To save the answers, provide a name and date (no spaces). Otherwise, type \'end\'")
if not name == 'end':
    save_answers_to_csv(name, base_chatbot_answers)

To save the answers, provide a name and date (no spaces). Otherwise, type 'end'gpt-3.5-turbo-1106


In [ ]:
answers_chatbot_answers_diff = []
chatbot_answers_base_chatbot_answers_diff = []
base_chatbot_answers_answers_diff = []

for x in questions:
    answers_chatbot_answers_diff.append(int(answers[x])-int(chatbot_answers[x]))
    chatbot_answers_base_chatbot_answers_diff.append(int(chatbot_answers[x])-int(base_chatbot_answers[x]))
    base_chatbot_answers_answers_diff.append(int(base_chatbot_answers[x])-int(answers[x]))

print(sum(answers_chatbot_answers_diff)/len(answers_chatbot_answers_diff))
print(sum(chatbot_answers_base_chatbot_answers_diff)/len(chatbot_answers_base_chatbot_answers_diff))
print(sum(base_chatbot_answers_answers_diff)/len(base_chatbot_answers_answers_diff))

-0.6319018404907976
0.2085889570552147
0.4233128834355828


Base vs. Tuned: 0.2085889570552147

Base vs. My answers: 0.4233128834355828

My answers vs. Tuned: 0.6319018404907976

